hack/generator/pkg/codegen/pipeline_augment_status.go

```diff
@@ -0,0 +1,421 @@
+					}
+				} else {
+					newTypes.Add(typeDef)
+				}
+			}
+
+			klog.Infof("Found status information for %v resources", found)
+			klog.Infof("Missing status information for %v resources", notFound)
+			klog.Infof("Input %v types, output %v types", len(types), len(newTypes))
+
+			return newTypes, nil
+		},
+	}
+}
+
+type resourceLookup map[astmodel.TypeName]astmodel.Type
+
+// makeStatusLookup makes a map of old name (non-status name) to status type
+func makeStatusLookup(swaggerTypes swaggerTypes) (resourceLookup, []astmodel.TypeDefinition) {
+	statusVisitor := makeStatusVisitor()
+
+	var otherTypes []astmodel.TypeDefinition
+	for typeName, typeDef := range swaggerTypes.types {
+		newName := appendStatusToName(typeName)
+		otherTypes = append(otherTypes, astmodel.MakeTypeDefinition(newName, statusVisitor.Visit(typeDef.Type(), nil)))
+	}
+
+	resources := make(resourceLookup)
+	for resourceName, resourceDef := range swaggerTypes.resources {
+		resources[resourceName] = statusVisitor.Visit(resourceDef.Type(), nil)
+	}
+
+	return resources, otherTypes
+}
+
+func appendStatusToName(typeName astmodel.TypeName) astmodel.TypeName {
+	return astmodel.MakeTypeName(typeName.PackageReference, typeName.Name()+"_Status")
+}
+
+func makeStatusVisitor() astmodel.TypeVisitor {
+	visitor := astmodel.MakeTypeVisitor()
+
+	visitor.VisitTypeName = func(this *astmodel.TypeVisitor, it astmodel.TypeName, ctx interface{}) astmodel.Type {
+		return appendStatusToName(it)
+		/*
+			if statusType, ok := statusTypes[it]; ok {
+				switch statusType.Type().(type) {
+				case *astmodel.ResourceType:
+				case *astmodel.ObjectType:
+					return appendStatusToName(it)
+				}
+			}
+
+			return it
+		*/
+	}
+
+	return visitor
+}
+
+var swaggerVersionRegex = regexp.MustCompile("\\d{4}-\\d{2}-\\d{2}(-preview)?")
+
+type swaggerTypes struct {
+	resources astmodel.Types
+	types     astmodel.Types
+}
+
+func loadSwaggerData(ctx context.Context, idFactory astmodel.IdentifierFactory, config *config.Configuration) (swaggerTypes, error) {
+
+	result := swaggerTypes{
+		resources: make(astmodel.Types),
+		types:     make(astmodel.Types),
+	}
+
+	cache := jsonast.MakeOpenAPISchemaCache()
+
+	for _, namespace := range config.Status.Schemas {
+		rootPath := path.Join(config.Status.SchemaRoot, namespace.BasePath)
+
+		// based off: https://github.com/Azure/azure-resource-manager-schemas/blob/1153d20f38e11bf1815bb855f309f83023fe6b0c/generator/cmd/listbasepaths.ts#L12-L15
+		if _, err := os.Stat(path.Join(rootPath, "readme.md")); os.IsNotExist(err) {
+			return swaggerTypes{}, errors.Errorf("configuration points to a directory that does not have status: %q", rootPath)
+		}
+
+		versionDirs, err := findVersionDirectories(rootPath)
+		if err != nil {
+			return swaggerTypes{}, errors.Wrapf(err, "unable to find version directories")
+		}
+
+		for _, versionDir := range versionDirs {
+			version := path.Base(versionDir)
+
+			schemas, err := loadSchemas(ctx, versionDir, cache)
```

Comment:
This is a bit wonky since I reworked it with the namespacing information from the config. Originally it was loading all the files in one pass; I'm going to restore it back to that model.